src/backend/replication/libpqwalreceiver/libpqwalreceiver.c

```diff
@@ -106,7 +107,45 @@ libpqrcv_connect(char *conninfo)
 			 "%s dbname=replication replication=true fallback_application_name=walreceiver",
 			 conninfo);
 
-	streamConn = PQconnectdb(conninfo_repl);
+	streamConn =  PQconnectStart(conninfo_repl);
+	if (PQstatus(streamConn) == CONNECTION_BAD)
```

Comment:
We should add `streamConn == NULL` check as well.

Reply:
`PQstatus` returns `CONNECTION_BAD` if `streamConn == NULL`